### ISSUE TITLE
Image url encode problem

### ISSUE THREAD

**Issue by norbertszivos** [jaimeiniesta/metainspector#215]
Hi,

Thank you your great gem.
Sadly we found an issue when we tried to use this gem.
We're using the latest version of this gem (5.4.1).

We found this issue when we tried to scraping theguardian pages. e.g. https://www.theguardian.com/politics/2017/sep/13/theresa-may-to-deliver-brexit-speech-in-florence but I guess you can reproduce the problem with any pages of theguardian.
<img width="1633" alt="metainspector" src="https://user-images.githubusercontent.com/1647831/30425255-e1727fd4-9948-11e7-86d2-aeb846bc5254.png">

We need the best match of the images which is coming from the `page.images.best` and we get that image url as we expected, but the url doesn't work.
<img width="1633" alt="signatureerror" src="https://user-images.githubusercontent.com/1647831/30425156-8b5bbd22-9948-11e7-8bdd-62e5409bd6e7.png">

The problem is coming from the url encoding. For example the `page.meta["og:image"]` gives us this url (which is working):
```
https://i.guim.co.uk/img/media/d63524a252711a991bfc7ebe1d065f221c71ca13/0_147_4209_2525/master/4209.jpg?w=1200&h=630&q=55&auto=format&usm=12&fit=crop&crop=faces%2Centropy&bm=normal&ba=bottom%2Cleft&blend64=aHR0cHM6Ly91cGxvYWRzLmd1aW0uY28udWsvMjAxNi8wNS8yNS9vdmVybGF5LWxvZ28tMTIwMC05MF9vcHQucG5n&s=e13bb69825bcc74d600294893dbef88e
```
but the `page.images.best` gives us this url (which is not working):
```
https://i.guim.co.uk/img/media/d63524a252711a991bfc7ebe1d065f221c71ca13/0_147_4209_2525/master/4209.jpg?w=1200&h=630&q=55&auto=format&usm=12&fit=crop&crop=faces,entropy&bm=normal&ba=bottom,left&blend64=aHR0cHM6Ly91cGxvYWRzLmd1aW0uY28udWsvMjAxNi8wNS8yNS9vdmVybGF5LWxvZ28tMTIwMC05MF9vcHQucG5n&s=e13bb69825bcc74d600294893dbef88e
```
So it's converting `%2C` to `,` and the cache server throw an error.

**Comment by jaimeiniesta** [comment]
Thanks for the report! I'll have a look as soon as possible, but feel free to send a pull request with a fix :)

**Comment by norbertszivos** [comment]
Hi @jaimeiniesta,
I'm happy to do that, but I'm not sure how to fix it at this moment, because there is a `normalize` method for the urls which cause the problem, but I guess this method is there for a reason. So it's not obvious where can we fix this problem or how.

**Comment by jaimeiniesta** [comment]
Thanks for looking into that @norbertszivos - I'll take a look as time permits.

**Comment by martijndekuijper** [comment]
Hey, any progress on this? Would be great if this could be fixed. Or is there a workaround?

**Comment by jaimeiniesta** [comment]
Still haven't had a chance to look at it. PRs are very welcome :)

**Comment by martijndekuijper** [comment]
@jaimeiniesta Unfortunately, I don't know how to fix this :) Any suggestions for a workaround?

**Comment by jaimeiniesta** [comment]
@martijndekuijper do you have more examples of this bug? Is the same happening in your case, `%2C` being converted to `, `? The more examples we have, the better. :)

**Comment by martijndekuijper** [comment]
@jaimeiniesta Yeah, good one. I experience the same thing with a Guardian article, but here's another example:

The url https://wip.chat/questions/194 has an Open Graph image with the following url:

`http://beta.url2png.com/v6/P53FB950639057/102aeef3a791fad1c2c52ffdb0b5897c/png/?fullpage=false&amp;thumbnail_max_width=1120&amp;unique=1517545798&amp;url=https%3A%2F%2Fwip.chat%2Fquestions%2F194.card&amp;viewport=1120x600`

`page.images.best` decodes the value of the `?url` parameter, but this should not happen. It does properly unescape the html entities though.

I hope this helps.

**Comment by jaimeiniesta** [comment]
Still failing on MetaInspector 5.6.0.

**Comment by jschwindt** [comment]
As far as I can see, removing URL normalization for image URLs would solve this problem. I'd like to contribute with a PR, what is your suggestion @jaimeiniesta: 1) remove normalization for `owner_suggested` only, 2) remove normalization for all image URLs, 3) take into account the gem `normalize_url` parameter and apply it to image URLs also, 4) create a new parameter `normalize_image_url`, 5) any other option?

**Comment by jaimeiniesta** [comment]
I think that what needs to be fixed is URL normalization (probably upstream at the Addressable gem), so that it does not break URLs like the ones above. I would not skip normalization, let's make it work for all these cases.

**Comment by jschwindt** [comment]
I'm not sure the normalization is doing something wrong, the problem is on those cache servers that look up the images using the query string literally. That's why I thought that leaving the URL as you find in the `meta['og:image']` could be an option.

**Comment by jaimeiniesta** [comment]
@jschwindt I understand, makes sense. Then yes, I think we can try to skip URL normalization in the images can be a fix to this, let's try that if you wish.
